### PR TITLE
sorting for NonEmptyList

### PR DESCRIPTION
**Pull request by matthewleon**
Adds the two sorting functions to `NonEmptyList`.

### CODE REVIEW

**Comment by paf31** [comment]
@garyb  I think we can make this a minor version bump, yes?

File: src/Data/List/NonEmpty.purs [hunk 1]
```diff
@@ -62,3 +65,10 @@ length (NonEmptyList (x :| xs)) = 1 + L.length xs
 
 concatMap :: forall a b. (a -> NonEmptyList b) -> NonEmptyList a -> NonEmptyList b
 concatMap = flip bind
+
+sort :: forall a. Ord a => NonEmptyList a -> NonEmptyList a
+sort xs = sortBy compare xs
+
+sortBy :: forall a. (a -> a -> Ordering) -> NonEmptyList a -> NonEmptyList a
+sortBy cmp xs = unsafeFromList $ L.sortBy cmp (toList xs)
```
Comment (paf31) [inline src/Data/List/NonEmpty.purs]: Another option here is to sort the tail and insert the head into the sorted result, a la insertion sort. That would require us to change the type of `insert` though, or to provide an alternative version.

**Comment by paf31** [comment]
@matthewleon Sorry for the delay, could you please rebase?

**Comment by garyb** [comment]
> I think we can make this a minor version bump, yes?

Agreed 👍

File: src/Data/List/NonEmpty.purs (hunk 1 above)
Comment (matthewleon) [inline src/Data/List/NonEmpty.purs]: @paf31 are you suggesting we should go down that route? I understand that `unsafeFromList` is, in many cases, a last resort, but here it seems to be both safe and more efficient than the alternative.

Comment (garyb) [inline src/Data/List/NonEmpty.purs]: I'd vote for the `unsafeFromList` version here too, personally. :)

**Comment by matthewleon** [comment]
> Sorry for the delay, could you please rebase?

done

**Comment by paf31** [comment]
Thanks!

**Pull request merged**